### DISCUSSION TITLE
Plugins API Docs

### DISCUSSION THREAD

**Discussion by LightBlueStuff24** [Acode-Foundation/Acode#452] [Q&A]
Is there any docs to create custom plugins?

**Comment by deadlyjack** [comment]
Yes, check out this [repository](https://github.com/deadlyjack/acode-plugin)

**Reply by LightBlueStuff24** [reply]
in dist the main.js file is too big & what should i do with that?

**Reply by deadlyjack** [reply]
dist/main.js is compiled code. You don't need to worry about it. You write your code in src/main.js

**Reply by LightBlueStuff24** [reply]
> dist/main.js is compiled code. You don't need to worry about it. You write your code in src/main.js

what the ***pop ()*** does?

**Reply by LightBlueStuff24** [reply]
> dist/main.js is compiled code. You don't need to worry about it. You write your code in src/main.js

also can i add custom snippets?

**Reply by clausnori** [reply]
> Yes, check out this [repository](https://github.com/deadlyjack/acode-plugin)

Hi ,I write plugins, but I can’t figure out how to upload files JSON
I use fs and .readfile(),but it doesn't work ,if the piece of code with the best option for loading files into the plugin?

**Reply by bajrangCoder** [reply]
> > Yes, check out this [repository](https://github.com/deadlyjack/acode-plugin)
> 
> Hi ,I write plugins, but I can’t figure out how to upload files JSON I use fs and .readfile(),but it doesn't work ,if the piece of code with the best option for loading files into the plugin?

What do you mean ? Provide some more context.
And also create a new discussion.

**Reply by clausnori** [reply]
sorry,no need now,I just imported via JS import
